tracer/src/Datadog.Trace/Debugger/SpanCodeOrigin/SpanCodeOriginManager.cs

```diff
@@ -5,94 +5,219 @@
+                        {
+                            try
+                            {
+                                var sequencePoint = reader.GetMethodSourceLocation(token);
+                                builder.Add(token, sequencePoint);
+                            }
+                            catch (Exception ex)
+                            {
+                                Log.Error(ex, "Failed to get sequence point for method token {Token} in assembly {AssemblyName}", property0: token, asm.FullName);
+                            }
+                        }
+
+                        return new AssemblyPdbInfo(builder.ToImmutable());
+                    }
+                    catch (Exception ex)
+                    {
+                        Log.Error(ex, "Failed to load PDB info for assembly {AssemblyName}", asm.Location);
+                        return null;
+                    }
+                });
+
+            return pdbInfo?.MethodSequencePoints.GetValueOrDefault<DatadogMetadataReader.DatadogSequencePoint?>(method.MetadataToken);
         }
 
-        private void AddExitSpanTag(Span span)
+        private void AddExitSpanTags(Span span)
         {
-            var frames = ArrayPool<FrameInfo>.Shared.Rent(_settings.CodeOriginMaxUserFrames);
+            var frames = ArrayPool<FrameInfo>.Shared.Rent(this._maxUserFrames);
             try
             {
                 var framesLength = PopulateUserFrames(frames);
                 if (framesLength == 0)
                 {
-                    Log.Warning("No user frames were found");
                     return;
                 }
 
-                span.Tags.SetTag($"{CodeOriginTag}.type", "exit");
                 for (var i = 0; i < framesLength; i++)
                 {
                     ref var info = ref frames[i];
 
                     // PopulateUserFrames returns only frames that have method
                     var method = info.Frame.GetMethod()!;
-                    var type = method.DeclaringType?.FullName ?? method.DeclaringType?.Name;
-                    if (string.IsNullOrEmpty(type))
+                    var typeName = method.DeclaringType?.FullName ?? method.DeclaringType?.Name;
+                    if (string.IsNullOrEmpty(typeName))
                     {
                         continue;
                     }
 
-                    span.Tags.SetTag($"{CodeOriginTag}.{FramesPrefix}.{i}.index", info.FrameIndex.ToString());
-                    span.Tags.SetTag($"{CodeOriginTag}.{FramesPrefix}.{i}.method", method.Name);
-                    span.Tags.SetTag($"{CodeOriginTag}.{FramesPrefix}.{i}.type", type);
+                    span.Tags.SetTag(_tags.Type, "exit");
+                    span.Tags.SetTag(_tags.Index[i], info.FrameIndex.ToString());
+                    span.Tags.SetTag(_tags.Method[i], method.Name);
+                    span.Tags.SetTag(_tags.TypeName[i], typeName);
```

Comment:
It was also probably flawed before this PR, but - aren't we lying about the index? If we skip certain methods in `PopulateUserFrames` (skipping all 3rd parties), won't it break the UI since we'll provide Code Origin for frame indices of those 3rd parties? Shall we also count them and use the correct indices here?